lib/nerves_time/hardware_time_module.ex

```diff
@@ -0,0 +1,55 @@
+defmodule NervesTime.HardwareTimeModule do
+  require Logger
+
+  @moduledoc """
+  Nerves Hardware Time Handler behaviour.
+
+  ```elixir
+  defmodule MyApp.MyRTCHandler do
+    @behaviour NervesTime.HardwareTimeModule
+  end
+  ```
+  """
+
+  @doc """
+  Return the stored timestamp from hardware time module.
+  """
+  @callback retrieve_time() :: NaiveDateTime.t() | :error
+
+  @doc """
+  Store current time.
+  """
+  @callback update_time() :: :ok | :error
+
+  @spec module() :: Module.t()
+  defp module() do
+    Application.get_env(:nerves_time, :hardware_time_module, NervesTime.FileTime)
```

Comment:
I'd like a list here so that you can change more than one option together. For example, `[MyRTC, NervesTime.FileTime]`. The RTC would be called first, but if it's unset, then FileTime can supply a time. `Set` would be called on all of them.

Reply:
That'd be easy to do, but not sure about the expected behavior.